extensions/_coresetup/init.lua

```diff
@@ -79,6 +79,44 @@ hs.relaunch = function()
     hs._exit(true, true)
 end
 
+--- hs.coroutineApplicationYield([delay])
+--- Function
+--- Yield coroutine to allow the Hammerspoon application to process other scheduled events and schedule a resume in the event application queue.
+---
+--- Parameters:
+---  * `delay` - an optional number, default `hs.math.minFloat`, specifying the number of seconds from when this function is executed that the `coroutine.resume` should be scheduled for.
+---
+--- Returns:
+---  * None
+---
+--- Notes:
+---  * this function will return an error if invoked outside of a coroutine.
+---  * unlike `coroutine.yield`, this function does not allow the passing of (new) information to or from the coroutine while it is running; this function is to allow long running tasks to yield time to the Hammerspoon application so other timers and scheduled events can occur without requiring the programmer to add code for an explicit resume.
+---
+---  * this function is added to the lua `coroutine` library as `coroutine.applicationYield` as an alternative name.
+local resumeTimers = {}
```

Comment:
Not sure exactly what this means... I need a pace to stick timers so they won't get collected; since this is referenced within the new function, it's an upvalue and won't get collected (my purpose) so... what's stickler's issue and/or how do I suppress this?

Reply:
Looks like a bug: https://github.com/mpeterv/luacheck/issues/156

You can use:

```lua
-- luacheck: ignore
```

...to ignore the error.

Reply:
FYI: Here's the LuaCheck documentation:

https://luacheck.readthedocs.io/en/stable/